src/Nethermind/Nethermind.Trie/Pruning/TrieStore.cs

```diff
@@ -757,40 +757,47 @@ private void PersistOnShutdown()
             {
                 // here we try to shorten the number of blocks recalculated when restarting (so we force persist)
                 // and we need to speed up the standard announcement procedure so we persists a block
-                // from the past (by going max reorg back)
 
-                BlockCommitSet? persistenceCandidate = null;
-                bool firstCandidateFound = false;
-                while (_commitSetQueue.TryDequeue(out BlockCommitSet? blockCommitSet))
+                List<BlockCommitSet> candidateSets = new();
+                while (_commitSetQueue.TryDequeue(out BlockCommitSet? frontSet))
                 {
-                    if (blockCommitSet is not null)
+                    if (candidateSets.Count == 0)
                     {
-                        if (firstCandidateFound == false)
+                        candidateSets.Add(frontSet);
+                    }
+                    else if (candidateSets[0].BlockNumber == frontSet!.BlockNumber)
+                    {
+                        candidateSets.Add(frontSet);
+                    }
+                    else
+                    {
+                        if (frontSet!.BlockNumber >= LatestCommittedBlockNumber - Reorganization.MaxDepth)
                         {
-                            persistenceCandidate = blockCommitSet;
-                            if (_logger.IsDebug) _logger.Debug($"New persistence candidate {persistenceCandidate}");
-                            firstCandidateFound = true;
                             continue;
                         }
-
-                        if (blockCommitSet.BlockNumber <= LatestCommittedBlockNumber - Reorganization.MaxDepth)
+                        else if (frontSet!.BlockNumber > candidateSets[0].BlockNumber)
                         {
-                            persistenceCandidate = blockCommitSet;
-                            if (_logger.IsDebug) _logger.Debug($"New persistence candidate {persistenceCandidate}");
+                            candidateSets = new();
```

Comment:
why recreate instead of clear?

Reply:
Hmm... don't know. No real reason i guess.